Sources/MapboxNavigation/DateComponentsFormatter+NavigationAdditions.swift

```diff
@@ -21,4 +21,14 @@ extension DateComponentsFormatter {
         formatter.allowedUnits = [.day, .hour, .minute]
         return formatter
     }()
+    
+    public static func travelTimeString(_ interval: TimeInterval,
+                                        signed: Bool,
+                                        unitStyle: DateComponentsFormatter.UnitsStyle?) -> String {
+        let formatter = DateComponentsFormatter()
+        formatter.unitsStyle = unitStyle ?? (interval < 3600 ? .short : .abbreviated)
```

Comment:
Can you elaborate on why this code uses `short` for seconds but `abbreviated` for minutes or longer?

Reply:
That's to keep the format the [same as we have](https://github.com/mapbox/mapbox-navigation-ios/blob/main/Sources/MapboxNavigation/BottomBannerViewController.swift#L209) for ETA display.

Reply:
Makes sense. In that case, let’s factor this DateComponentsFormatter instance into a `travelTimeDateComponentsFormatter` constant on this extension so that `updateETA(routeProgress:)` can use it too.

Reply:
`unitStyle` has to be updated quite often, so we won't get away with specially configured instance. I've extracted unit selection into a method instead.